sklearn/model_selection/tests/test_search.py

```diff
@@ -2686,3 +2685,39 @@ def score(self, X, y):
     grid_search.fit(X, y)
     for param in param_grid:
         assert grid_search.cv_results_[f"param_{param}"].dtype == object
+
+
+@pytest.mark.filterwarnings(
+    "ignore:in the future the `.dtype` attribute of a given datatype object must "
+    "be a valid dtype instance:DeprecationWarning"
```

Comment:
who's rasing this? As in, are the users gonna see this now?

Reply:
NumPy raises it in the line `np.result_type(*param_list)`

It's a DeprecationWarning, so it wouldn't ordinarily be visible to end users, which is why running the example in the linked issue doesn't show any warning https://github.com/scikit-learn/scikit-learn/issues/29157

Still, doesn't hurt to silence it, I've gone with that 👍